narwhal/primary/src/state_handler.rs

```diff
@@ -447,6 +437,57 @@ impl RandomnessState {
                 return;
             }
         };
+
+        // Try to verify the aggregated signature all at once. (Should work in the happy path.)
+        if ThresholdBls12381MinSig::verify(dkg_output.vss_pk.c0(), &round.signature_message(), &sig)
+            .is_err()
+        {
+            // If verifiation fails, some of the inputs must be invalid. We have to go through
+            // one-by-one to find which.
+            self.partial_sigs.retain(|&(r, authority_id), partial_sig| {
+                if ThresholdBls12381MinSig::partial_verify_batch(
```

Comment:
let's add a todo to have a test for this flow?

Reply:
done